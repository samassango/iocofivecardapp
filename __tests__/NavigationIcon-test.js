import 'react-native';
import React from 'react';
import NavigationIcon from '../src/components/navigationicon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const navIconTree = renderer.create(<NavigationIcon icon={"navigate-before"} onEventTouch={() => console.log('navigate-before')} />);

it('renders navigationicon component correctly', () => {
  expect(navIconTree).toMatchSnapshot();
});

it('Check prop icon matches', () => {
  expect(navIconTree.root.props.icon).toBe("navigate-before");
});

it('Check prop onEventTouch matches', () => {
  expect(navIconTree.root.props.onEventTouch).toBeTruthy();
});
