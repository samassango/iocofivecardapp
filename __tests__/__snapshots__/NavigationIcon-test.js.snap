// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navigationicon component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 667,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="arrow-navigate-before"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
